src/argilla/client/feedback/dataset/base.py

```diff
@@ -112,20 +109,31 @@ def question_by_name(self, name: str) -> Optional[Union[AllowedQuestionTypes]]:
 
     @property
     @abstractmethod
-    def metadata_properties(self) -> Union[List["AllowedMetadataPropertyTypes"]]:
+    def metadata_properties(
+        self,
+    ) -> Union[List["AllowedMetadataPropertyTypes"], List["AllowedRemoteMetadataPropertyTypes"]]:
         """Returns the metadata properties that will be indexed and could be used to filter the dataset."""
         pass
 
-    def metadata_property_by_name(self, name: str) -> Optional[Union["AllowedMetadataPropertyTypes"]]:
+    def metadata_property_by_name(
+        self, name: str
+    ) -> Optional[Union["AllowedMetadataPropertyTypes", "AllowedRemoteMetadataPropertyTypes"]]:
```

Comment:
Why making this type hint more complex? The remote metadata properties classes inherits from the non-remote ones so the type hint was correct.

Reply:
Same in the rest of methods